apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerTransaction.cs

```diff
@@ -353,24 +353,29 @@ Task<IEnumerable<long>> ISpannerTransaction.ExecuteBatchDmlAsync(ExecuteBatchDml
         /// Commits the database transaction synchronously, returning the database timestamp for the commit via <paramref name="timestamp"/>.
         /// </summary>
         /// <param name="timestamp">Returns the UTC timestamp when the data was written to the database.</param>
-        public void Commit(out DateTime timestamp) => timestamp = Task.Run(() => CommitAsync(default)).ResultWithUnwrappedExceptions();
+        public void Commit(out DateTime timestamp) => timestamp = Task.Run(() => CommitAndReturnTimestampAsync(default)).ResultWithUnwrappedExceptions();
 
-        // FIXME: This is "new" because there's now CommitAsync in the BCL. That returns just Task, rather than Task<DateTime>,
-        // which is awkward...        
+        /// <inheritdoc />
+        public
+#if NETSTANDARD2_1_OR_GREATER
+            override
+#endif
+            Task CommitAsync(CancellationToken cancellationToken = default) => CommitAndReturnTimestampAsync(cancellationToken);
 
         /// <summary>
-        /// Commits the database transaction asynchronously.
+        /// Commits the database transaction asynchronously, returning the UTC timestamp when the data was written to the database.
         /// </summary>
         /// <param name="cancellationToken">A cancellation token used for this task.</param>
         /// <returns>Returns the UTC timestamp when the data was written to the database.</returns>
-#if NET462
-        public Task<DateTime> CommitAsync(CancellationToken cancellationToken = default)
-#else
-        public new Task<DateTime> CommitAsync(CancellationToken cancellationToken = default)
-#endif
+        public Task<DateTime> CommitAndReturnTimestampAsync(CancellationToken cancellationToken = default)
```

Comment:
So that we don't have to add the breaking change nor this method I think we can add an intermediate internal class, `SpannerTransactionImpl` that overrides `CommitAsync` for asynchrony and then we can have the public `SpannerTransaction` hide the `CommitAsync` implementation to change the return type.

What do you think?

Reply:
I did wonder about that. https://docs.microsoft.com/en-us/dotnet/core/compatibility/ states that it's a "requires judgement" change - we can probably be careful enough, and it's across a major boundary anyway.

Will try this in a new PR.

Reply:
Alternative approach is now in https://github.com/googleapis/google-cloud-dotnet/pull/8535 - let me know if that's not what you meant.